Tests/UnitTests/Networking/HTTPClientTests.swift

```diff
@@ -281,33 +315,67 @@ class HTTPClientTests: XCTestCase {
         }
 
-        self.client.perform(request, authHeaders: [:]) {
-            result.value = $0
+        self.client.perform(request, authHeaders: [:]) { (response: HTTPResponse<CustomResponse>.Result) in
+            result.value = response
         }
 
         expect(result.value).toEventuallyNot(beNil(), timeout: .seconds(1))
         expect(result.value).to(beFailure())
+
+        // TODO: custom error?
+        let error = try XCTUnwrap(result.value?.error as NSError?)
+        expect(error.domain) == RCPurchasesErrorCodeDomain
+        expect(error.code) == ErrorCode.apiEndpointBlockedError.rawValue
     }
 
     func testServerSide200s() {
         let request = HTTPRequest(method: .get, path: .mockPath)
 
-        let result: Atomic<Result<HTTPResponse, Error>?> = .init(nil)
+        let responseData = "{\"message\": \"something is great up in the cloud\"}".data(using: String.Encoding.utf8)!
+
+        let result: Atomic<Result<HTTPResponse<Data>, Error>?> = .init(nil)
 
         stub(condition: isPath(request.path)) { _ in
-            let json = "{\"message\": \"something is great up in the cloud\"}"
-            return HTTPStubsResponse(data: json.data(using: String.Encoding.utf8)!,
+            return HTTPStubsResponse(data: responseData,
                                      statusCode: .success,
                                      headers: nil)
         }
 
-        self.client.perform(request, authHeaders: [:]) {
-            result.value = $0
+        self.client.perform(request, authHeaders: [:]) { (response: HTTPResponse<Data>.Result) in
+            result.value = response
         }
 
         expect(result.value).toEventuallyNot(beNil(), timeout: .seconds(1))
         expect(result.value).to(beSuccess())
-        expect(result.value?.value?.jsonObject["message"] as? String) == "something is great up in the cloud"
+        expect(result.value?.value?.body) == responseData
+    }
+
+    func testResponseDeserialization() throws {
+        struct CustomResponse: Codable, Equatable, HTTPResponseBody {
+            let message: String
+        }
+
+        let request = HTTPRequest(method: .get, path: .mockPath)
+
+        let response = CustomResponse(message: "Something is great up in the cloud")
+        let responseData = try JSONEncoder.default.encode(response)
+
+        let result: Atomic<Result<HTTPResponse<CustomResponse>, Error>?> = .init(nil)
+
+        stub(condition: isPath(request.path)) { _ in
+            return HTTPStubsResponse(data: responseData,
+                                     statusCode: .success,
+                                     headers: nil)
+        }
+
+        self.client.perform(request, authHeaders: [:]) { (response: HTTPResponse<CustomResponse>.Result) in
+            result.value = response
+        }
+
+        expect(result.value).toEventuallyNot(beNil(), timeout: .seconds(1))
+        expect(result.value).to(beSuccess())
+        expect(result.value?.value?.body) == response
```

Comment:
The main test in the `HTTPClient` change, the body becomes the deserialized `CustomResponse` 🎉